query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -271,8 +271,21 @@ private String convertExpression2String(
             viewDefinition.getPhysicalColumnNames(getLogicalColumnName(expression));
         return joiner.join(columnNames);
       case ATTRIBUTE_EXPRESSION:
-        if (isComplexAttribute(expression)) {
-          /** TODO:Handle complex attribute */
+        if (isMapSubpathAttributeExpression(expression, viewDefinition)) {
```

Comment:
@aaron-steinfeld based on discussions,
here we will add validation that any attribute expression with a subpath should always be a map attribute.

Reply:
Non-complex attributes having subPath should error out with `IllegalArgumentException`.
The test example `span.id` with `subPath` in MigrationTest.

Reply:
The main blocker as of now is to decide what to do with a bad request (having subpath and not being a map attribute) like :
```
AttributeExpression {
attributeId: API_TRACE.startTime";
subpath: "1234"
}
```

Ronak had suggested adding a validation here (I think filter would be covered here too). In validation, we will throw exceptions for such cases. 
The validation would be to check directly from the view definition. Then we can remove the below function from everywhere :
```
  private static boolean isMapField(String columnName, ViewDefinition viewDefinition) {
    return viewDefinition.getColumnType(columnName) == ValueType.STRING_MAP;
  }
```

What do you think ?

Reply:
Validation shouldn't be in the converter - it should be in the existing validate method before we hit the converter.

Reply:
Is there any validator as of now in QS ? I couldn't find one.

Reply:
It's minimal (we usually just let pinot do it), but the existing validation is in `PinotBasedRequestHandler.validateQueryRequest`

Reply:
I guess we need only one validation that attribute expression with subpath should be map attribute. This will resolve most of the conversations. Should I add this in `validateQueryRequest` function ?

Reply:
This will also help in naming all the util functions appropriately.

Reply:
resolved